qualang_tools/virtual_gates/virtual_gate_sequence.py

```diff
@@ -0,0 +1,252 @@
+import numpy as np
+
+from qm.qua._dsl import QuaVariable, QuaExpression
+from qm.qua import declare, assign, play, fixed, Cast, amp, wait, ramp, ramp_to_zero
+from typing import Union, List, Dict
+
+
+class VoltageGateSequence:
+    def __init__(self, configuration: Dict, elements: List[str]):
+        """
+        Initializes a VirtualGateSequence object for designing arbitrary pulse sequences using virtual gates.
+
+        This class allows the creation of complex pulse sequences by defining voltage levels (points) in gate-space
+        and durations for multiple elements. It supports the addition of steps and ramps, and keeps track of average
+        voltage levels for compensation pulses.
+
+        The `configuration` provided will be updated to include necessary operations and waveforms for the sequence.
```

Comment:
Which means that this class must be called before opening the QM, right? Should be mentioned (probably in the readme file).
Or even more, that the program itself should be written before opening the qm

Reply:
Added a note in the README, please check